firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigAutoFetch.java

```diff
@@ -106,32 +119,43 @@ public void listenForNotifications() {
     try {
       scheduledExecutorService.awaitTermination(3L, TimeUnit.SECONDS);
     } catch (InterruptedException ex) {
-      Log.i(TAG, "Thread Interrupted.");
+      Log.d(TAG, "Thread Interrupted.");
     }
   }
 
   // Auto-fetch new config and execute callbacks on each new message
   private void handleNotifications(InputStream inputStream) throws IOException {
     BufferedReader reader = new BufferedReader((new InputStreamReader(inputStream, "utf-8")));
     String message;
+    String fullMessage = "";
     while ((message = reader.readLine()) != null) {
-      if (!message.contains("latestTemplateVersionNumber")) {
-        continue;
-      }
+      fullMessage += message;
```

Comment:
I'm having trouble following the parsing logic. Could we parse the JSON response first, then get the fields we need? Example in fetch: https://github.com/firebase/firebase-android-sdk/blob/c6cc8c5d4a00518033a6f281486c9610b821e1c4/firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigFetchHttpClient.java#L372-L383

Reply:
For our[ parsing](https://github.com/firebase/firebase-android-sdk/blob/6150f7b28cbeff3ba81d98d29bc6419b1a1cd43c/firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigAutoFetch.java#L117), since it's a stream there will be multiple messages coming through our while loop so we can't just end it when we get one message. So what we do is add every incoming message to our string fullMessage string. When we see a message that contains '}' then we know the backend just finished sending a full message. We then call parseMessage which just strips the beginning and end of fullMessage up until it reaches '{' and '}'. Everything between '{' and. '}' should be properly formatted so we don't need to do anything there.